Branch-SDK/Branch-SDK/BNCCallbacks.h

```diff
@@ -17,5 +17,6 @@ typedef void (^callbackWithStatus) (BOOL changed, NSError * _Nullable error);
 typedef void (^callbackWithList) (NSArray * _Nullable list, NSError * _Nullable error);
 typedef void (^callbackWithUrlAndSpotlightIdentifier) (NSString * _Nullable url, NSString * _Nullable spotlightIdentifier, NSError * _Nullable error);
 typedef void (^callbackWithBranchUniversalObject) (BranchUniversalObject * _Nullable universalObject, BranchLinkProperties * _Nullable linkProperties, NSError * _Nullable error);
+typedef void (^completion) (NSError * __nullable error);
```

Comment:
If callbacks are defined with typedefs then autocompletion breaks, and partners need to look up the definition each time.

Reply:
Another thought here: These define globally-visible types which can potentially collide with app code. And in Swift, all types are capitalized, even arithmetic types like Bool, Int, Double. But these types show up with lower-case names, which at least looks odd and unidiomatic when you have to make use of them. It might be better if these had names like BranchCallbackWithList, etc. And a simple symbol like `completion` is almost guaranteed to conflict with something somewhere.

Not using typedefs for something this simple is also beneficial, as Ed points out. There's a lot to be said for using literals in some cases. It makes things easier to use.